gitpod-io/gitpod
# Ensure uncommitted changes are displayed in dashboard

## Description
Ensure uncommitted changes are displayed in dashboard. When the workspace was restored from a backup and subsequently stopped, uncommitted changes were not displayed in the dashboard anymore. The reason was that the location we use to check for changes was incorrect, because we get the checkout location from the initializer and the backup initializer does not use git and hence cannot provide the checkout location.

The checkout location is now provided as part of the workspace init request.

## Related Issue(s)
Fixes #7370

## How to test
- Create new workspace
- Modify or create a new file
- Stop the workspace 
- Navigate to dashboard -> changes are displayed for the workspace
- Start the workspace again
- Stop the workspace
- Navigate to dashboard -> changes should still be displayed for the workspace

## Notes for reviewers
It looks like bigger repositories like gitpod itself currently do not backup correctly in the preview environment (unreleated to my changes). I used [this](https://github.com/Furisto/brecht) for testing.

## Release Notes
```release-note
Ensure uncommitted changes are displayed in dashboard when workspace was restored from backup
```
